wdn/templates_5.0/js-src/notice.babel.js

```diff
@@ -0,0 +1,331 @@
+let notices = document.querySelectorAll('[data-widget="notice"]');1111
+let fixedBottomExists = false; // flag for checking if a fixed bottom has been initialized on the page
+fixedBottomExists = document.querySelector('[id*="unl-widget-fixedBottom-"]') ? true : false; // if a fixedBottom notice has been previously added to the page set flag to true
+
+
+notices = [].slice.call(notices);
+
+// standard classes based on what kind of notice, keep it to DCF classes for now
+const noticeClasses  = {
+	notify: ['dcf-notice', 'unl-notice-notify'],
+	success: ['dcf-notice', 'unl-notice-success'],
+	alert: ['dcf-notice', 'unl-notice--alert'],
+	fatal: ['dcf-notice', 'unl-notice--fatal'],
+};
+
+const noticeLocationClasses = {
+	current: ['uno'],
+	nav: ['foo'],
+	fixedBottom:['bar', 'dcf-fixed', 'dcf-notice-fixedBottom', 'dcf-pin-bottom', 'dcf-pin-right', 'dcf-pin-left'],
+	fixedBottomLeft:['zoink'] // TODO add option that goes 50% width on desktop when fixedBottom
+};
+
+// default animations depending on associated locations
+// fixedBottom associated with slideUp animation. Everything else uses slideInScroll
+const noticeAnimationClasses = {
+	slideInScroll: ['baz'],
+	slideUp: ['car'],
+};
+
+const closeButtonClasses = ['dcf-absolute', 'dcf-pin-top', 'dcf-pin-right', 'dcf-mt-3', 'dcf-mr-3' ,'dcf-btn', 'dcf-btn-tertiary', 'js-notice-toggle'];
+
+/**
+ *
+ * Functions
+ */
+// function to generate GUID or UIDs
+function uuidv4() {
+	return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+		var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+		return v.toString(16);
+	});
+}
+
+/**
+ *
+ * @param el takes an element and move it to be first child of main
+ */
+function moveElement(el) {
+	const main = document.querySelector('main');
+	const firstChild = main.firstElementChild;
+	main.insertBefore(el, firstChild);
+}
+
+/**
+ * @purpose - permanently closes the notice element
+ * @param notice - notice to be closed
+ *
+ */
+function closeNotice(notice) {
+	notice.classList.add('dcf-notice-fixedBottom--close');
+	notice.addEventListener('transitionend', e => {
+		if (e.propertyName !== "max-height") return;
+		el.classList.add('dcf-d-none');
+		el.setAttribute('aria-hidden', 'true');
+		console.log(e);
+	});
+	localStorage.setItem(notice.id,'closed');
+}
```

Comment:
What element has focus when a notice is closed?

Reply:
As discussed, we will be sending focus to `<main>` after the notice closes.